### ISSUE TITLE
List and List Group Header a11y review

### ISSUE THREAD

**Issue by InnaAtanasova** [SAP/fundamental-ngx#14179]
List Group Header should be linked to the items of the group it belongs to. It should also have proper aria attributes (for example: missing aria-roledescription="Group Header").
The List items that belong to the group should have aria-setsize and aria-posinset.
